force-app/main/default/classes/Application_TEST.cls

```diff
@@ -0,0 +1,151 @@
+     ****************************************************** UNIT TESTS *********************************************************
+     **************************************************************************************************************************/
+
+    /********************************************************
+     * @description Should fail to insert an application with a due date prior to the open date
+     *********************************************************/
+    @isTest
+    static void shouldFailInsertWhenApplicationDueDateIsBeforeOpenDate() {
+        Account academicAccount = getAcademicAccount();
+        Contact student = getStudent();
+
+        Application__c invalidApplication = new Application__c(
+            Applicant__c = student.Id,
+            Applying_To__c = academicAccount.Id,
+            Due_Date__c = System.today().addDays(-10),
+            Open_Date__c = System.today()
+        );
+
+        Test.startTest();
+        Database.SaveResult insertResult = Database.insert(invalidApplication, false);
+        Test.stopTest();
+
+        System.assertEquals(false, insertResult.isSuccess(), 'Insert should have failed due to invalid date range');
+        System.assertEquals(
+            Label.stgErrorAppDueDate,
+            insertResult.getErrors()[0].getMessage(),
+            'Should fail with the Open/Due Date validation error message'
+        );
+    }
+
+    /********************************************************
+     * @description Should fail to update an application when setting the due date prior to the open date
+     *********************************************************/
+    @isTest
+    static void shouldFailUpdateWhenApplicationDueDateIsBeforeOpenDate() {
+        Account academicAccount = getAcademicAccount();
+        Contact student = getStudent();
+
+        Application__c openEndedApplication = new Application__c(
+            Applicant__c = student.Id,
+            Applying_To__c = academicAccount.Id,
+            Open_Date__c = System.today()
+        );
+        insert openEndedApplication;
+
+        Test.startTest();
+        openEndedApplication.Due_Date__c = System.today().addDays(-10);
+        Database.SaveResult updateResult = Database.update(openEndedApplication, false);
+        Test.stopTest();
+
+        System.assertEquals(false, updateResult.isSuccess(), 'Update should have failed due to invalid date range');
+        System.assertEquals(
+            Label.stgErrorAppDueDate,
+            updateResult.getErrors()[0].getMessage(),
+            'Should fail with the Open/Due Date validation error message'
+        );
+    }
+
+    /********************************************************
+     * @description Should succeed to insert an application with just one date set
+     *********************************************************/
+    @isTest
+    static void shouldSucceedWhenApplicationHasEitherDueOrOpenDate() {
```

Comment:
This test has a mixed context and is testing two different things. I would prefer that it test the individual behaviors separately so we have explicit and concise tests indicating exactly what behavior is functioning or failing.

This is not a required change, but I feel it would increase Trust.